### DISCUSSION TITLE
Connect to multiple databases

### DISCUSSION THREAD

**Discussion by dattatre263** [PostgREST/postgrest#2784] [Q&A] [answered]
Is it possible to connect to multiple databases ? I have a microservice architecture and each service connects to a seperate database in the same database server. Therefore I could have a connection string like so 

"postgres://user1:password1@localhost:5432/db1"  AND
"postgres://user2:password2@localhost:5432/db2"

Is this possible ?

**Comment by steve-chavez** [comment, marked as answer]
Not possible right now with a single instance. You'd have to run multiple postgrests.

There was some discussion about mapping a different db connection using the `Host` header but just that.

**Edit**: https://github.com/PostgREST/postgrest/issues/2798

---

Tangentially related, but libpq 16 is bringing load balancing with multiple dbs: https://mydbops.wordpress.com/2023/05/07/postgresql-16-brings-load-balancing-support-in-libpq-psql/

I wonder if libpq is really the right place to build connection pooling over.

**Comment by aadrian** [comment]
@dattatre263 If one has controll over the DB configuration, by using a foreign data wrapper, one can access more "non-local" resources, and also indirectly (e.g. with views/matviews - since PG has no synonims) expose them with PostgREST.
